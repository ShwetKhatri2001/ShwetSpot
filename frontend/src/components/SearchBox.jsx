import React, { useState } from 'react';
import { Button, Form } from 'react-bootstrap';

const SearchBox = ({ history }) => {
  const [keyword, setKeyword] = useState('');

  const submitHandler = e => {
    e.preventDefault();
    if (keyword.trim()) {
      history.push(`/search/${keyword}`);
    } else {
      history.push('/');
    }
  };

  return (
    <Form onSubmit={submitHandler} className="mx-auto d-inline-flex">
      <Form.Control
        id="search-box"
        type="text"
        name="q"
        onChange={e => setKeyword(e.target.value)}
        placeholder="Search Products..."
        className="mt-3"
      />
      <Button type="submit" variant="success" id="search-box-btn">
        Search
      </Button>
    </Form>
  );
};

export default SearchBox;
