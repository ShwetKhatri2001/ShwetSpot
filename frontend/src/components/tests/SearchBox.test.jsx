import React from 'react';
import renderer from 'react-test-renderer';

import SearchBox from '../SearchBox';
import { MemoryRouter } from 'react-router-dom';

it('should render correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <SearchBox />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
