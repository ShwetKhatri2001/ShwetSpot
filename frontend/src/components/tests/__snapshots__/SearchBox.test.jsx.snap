// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<form
  className="mx-auto d-inline-flex"
  onSubmit={[Function]}
>
  <input
    className="mt-3 form-control"
    id="search-box"
    name="q"
    onChange={[Function]}
    placeholder="Search Products..."
    type="text"
  />
  <button
    className="btn btn-success"
    disabled={false}
    id="search-box-btn"
    type="submit"
  >
    Search
  </button>
</form>
`;
